src/Event/UnsupportedEvent.php

```diff
@@ -0,0 +1,20 @@
+<?php declare(strict_types=1);
+/*
+ * This file is part of PHPUnit.
+ *
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace PHPUnit\Event;
+
+use PHPUnit\Exception;
+
+final class UnsupportedEvent extends \Exception implements Exception
+{
+    public static function type(Type $type): self
+    {
+        return new self(\sprintf('Type "%s" not supported', $type->asString()));
```

Comment:
I would add the class name of the subscriber here to create a bit of context: `Event subscriber "Foo\Bar\BazSubscriber" does not support events of type "blah"`

Reply:
Good idea.

Even though this should technically be almost impossible to ever trigger, given that the dispatcher uses the same method to find out what types of events to send to this subscriber.

Reply:
Good point, thank you, @rpkamp!